src/inputs/internal/ListBoxSection.tsx

```diff
@@ -0,0 +1,51 @@
+import { Node } from "@react-types/shared";
+import React from "react";
+import { useListBoxSection, useSeparator } from "react-aria";
+import { SelectState } from "react-stately";
+import { Css } from "src/Css";
+import { persistentItemHeight, sectionSeparatorHeight } from "src/inputs/internal/constants";
+import { Option } from "src/inputs/internal/Option";
+import { VirtualizedOptions } from "src/inputs/internal/VirtualizedOptions";
+
+interface ListBoxSectionProps<O> {
+  section: Node<O>;
+  state: SelectState<O>;
+  contrast: boolean;
+  onListHeightChange: (height: number) => void;
+  popoverHeight: number;
+  scrollOnFocus?: boolean;
+}
+export function ListBoxSection<O>(props: ListBoxSectionProps<O>) {
+  const { section, state, contrast, onListHeightChange, popoverHeight, scrollOnFocus } = props;
+  const { itemProps, groupProps } = useListBoxSection(section);
+  const { separatorProps } = useSeparator({ elementType: "li" });
+  const isPersistentSection = section.key !== state.collection.getFirstKey();
+  const childNodes = [...section.childNodes];
+
+  return (
+    <>
+      {isPersistentSection && <li {...separatorProps} css={Css.bt.bGray200.$} />}
+      <li {...itemProps} css={Css.if(!isPersistentSection).overflowAuto.$}>
+        {/* Styles assume only one Persistent Item is ever shown. Will need to adjust if that ever changes */}
+        <ul
+          css={
+            Css.listReset.if(!isPersistentSection).hPx(popoverHeight - sectionSeparatorHeight - persistentItemHeight).$
+          }
+          {...groupProps}
+        >
+          {isPersistentSection ? (
```

Comment:
Kinda wonder if separate `OptionSection` and `PersistentActionSection` functions would be simpler than the conditionals in this combined function? Maybe not though.

Reply:
Yea, I like the thought. I wanted to make a single generic ListBoxSection, that can be expanded upon in the future. But maybe we'd be better off being a bit more specific in the first go-round. I'm going to ponder that bit more and might create a follow up PR.